tools/build_executable.py

```diff
@@ -0,0 +1,108 @@
+    parser.add_argument("--config_mode", default="Shipping")
+    parser.add_argument("--num_parallel_jobs", type=int, default=1)
+    parser.add_argument("--output_dir", default=os.path.dirname(os.path.realpath(__file__)))
+    parser.add_argument("--temp_dir", default=os.path.join(os.path.dirname(os.path.realpath(__file__)), "tmp"))
+    parser.add_argument("--conda_script")
+    parser.add_argument("--commit_id")
+    args = parser.parse_args()
+
+    repo_dir           = os.path.join(os.path.realpath(args.temp_dir), "spear")
+    unreal_project_dir = os.path.realpath(os.path.join(repo_dir, "cpp", "unreal_projects", "SpearSim"))
+    unreal_plugins_dir = os.path.realpath(os.path.join(repo_dir, "cpp", "unreal_plugins"))
+
+    # remove any contents already present for a clean build
+    if os.path.exists(repo_dir):
+        shutil.rmtree(repo_dir)
+        os.makedirs(repo_dir)
+
+    if sys.platform == "win32":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.bat")
+        target_platform = "Win64"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+        cmd_prefix = f"conda activate {args.conda_env}&"
+    elif sys.platform == "darwin":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Mac"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}-Unsigned")
+        conda_script = args.conda_script
+        if not conda_script:
+            conda_script = "~/anaconda3/etc/profile.d/conda.sh"
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env};"
+    elif sys.platform == "linux":
+        run_uat_script = os.path.join(args.unreal_engine_dir, "Engine", "Build", "BatchFiles", "RunUAT.sh")
+        target_platform = "Linux"
+        archive_dir = os.path.join(args.output_dir, f"SpearSim-{target_platform}-{args.config_mode}")
+        conda_script = args.conda_script
+        if not conda_script:
+            conda_script = "~/anaconda3/etc/profile.d/conda.sh"
+        cmd_prefix = f". {conda_script}; conda activate {args.conda_env};"
+    else:
+        assert False
+
+    # clone repo along with submodules
+    cmd = ["git", "clone", "--recurse-submodules", "https://github.com/isl-org/spear", repo_dir]
+    print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd, check=True)
+    assert cmd_result.returncode == 0
+
+    if args.commit_id:
+        cmd = ["git", "reset", "--hard", args.commit_id]
+        print(f"[SPEAR | build_executable.py] Executing: {' '.join(cmd)}")
+        cmd_result = subprocess.run(cmd, check=True)
+        assert cmd_result.returncode == 0
+
+    # build third-party libs
+    cmd = ["python", "build_third_party_libs.py", "--num_parallel_jobs", f"{args.num_parallel_jobs}"]
```

Comment:
Why are we storing cmd as a list sometimes and as a string other times? Whatever we do in this file, we should be consistent.

Reply:
When we execute with `shell=True`, it needs to be a string, otherwise it should be a list.

Reply:
Ah ok got it.